### PR TITLE
docs: snippets of "Grouping routes for RPC"

### PR DESCRIPTION
**Pull request by sor4chi**

### CODE REVIEW

**Comment by yusukebe** [comment]
Hi @sor4chi 

I think this snippet is better to understand it:

```ts
const authorsApp = new Hono()
  .get('/', (c) => c.jsonT({ result: 'list authors' }))
  .post('/', (c) => c.jsonT({ result: 'create an author' }, 201))
  .get('/:id', (c) => c.jsonT({ result: `get ${c.req.param('id')}` }))

const booksApp = new Hono()
  .get('/', (c) => c.jsonT({ result: 'list books' }))
  .post('/', (c) => c.jsonT({ result: 'create a book' }, 201))
  .get('/:id', (c) => c.jsonT({ result: `get ${c.req.param('id')}` }))

const app = new Hono().route('/authors', authorsApp).route('/books', booksApp)

type AppType = typeof app
```

**Comment by sor4chi** [comment]
@yusukebe fixed!

File: snippets/grouping-routes-rpc.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Grouping routes for RPC
+
+To enable type inference for Hono Client on the segmented `app` using `app.route()`, you can do the following:
```
Comment (yusukebe) [inline snippets/grouping-routes-rpc.md:3]: Please rewrite to this:

```md
If you want to enable type inference for multiple `app`s correctly, you can use `app.route()` as follows:
```

File: snippets/grouping-routes-rpc.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+## Snippets
+
+Pass the return value of methods like `app.get()` or `app.post()` as Routes to `app.route()`.
```
Comment (yusukebe) [inline snippets/grouping-routes-rpc.md:7]: Please rewrite to this:

```md
Pass the value returned from methods like `app.get()` or `app.post()` to the second argument of `app.route()`.
```

File: snippets/grouping-routes-rpc.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  .post("/", (c) => c.jsonT({ result: "create a book" }, 201))
+  .get("/:id", (c) => c.jsonT({ result: `get ${c.req.param("id")}` }));
+
+const app = new Hono().route("/authors", authorsApp).route("/books", booksApp);
+
+type AppType = typeof app;
+
+const client = hc<AppType>("[YOUR HONO SERVER ENDPOINT]");
+
+async function main() {
```
Comment (yusukebe) [inline snippets/grouping-routes-rpc.md:29]: Maybe this line and those following it are not necessary.

**Review by yusukebe — COMMENTED** [review]
Thanks. I've left the comments.

**Comment by sor4chi** [comment]
@yusukebe Thanks, I fixed :-D

**Comment by yusukebe** [comment]
Thanks!

**Pull request merged**